scripts/build_base_energy_totals.py

```diff
@@ -0,0 +1,382 @@
+    if "snakemake" not in globals():
+        from helpers import mock_snakemake, sets_path_to_root
+
+        os.chdir(os.path.dirname(os.path.abspath(__file__)))
+
+        snakemake = mock_snakemake(
+            "build_base_energy_totals",
+            simpl="",
+            clusters=10,
+            demand="DF",
+            planning_horizons=2030,
+        )
+        sets_path_to_root("pypsa-earth-sec")
+
+    energy_stat_database = pd.read_excel(
+        snakemake.input.unsd_paths, index_col=0, header=0
+    )  # pd.read_excel("/nfs/home/haz43975/pypsa-earth-sec/scripts/Energy_Statistics_Database.xlsx"
+
+    # Load the links and make a dictionary
+    df = energy_stat_database.copy()
+    df = df.dropna(axis=0, subset=["Link"])
+    df = df.to_dict("dict")
+    d = df["Link"]
+
+    if snakemake.config["demand_data"]["update_data"]:
+        # Delete and existing files to avoid duplication and double counting
+
+        files = glob.glob("data/demand/unsd/data/*.txt")
+        for f in files:
+            os.remove(f)
+
+        # Feed the dictionary of links to the for loop, download and unzip all files
+        for key, value in d.items():
+            zipurl = value
+
+            with urlopen(zipurl) as zipresp:
+                with ZipFile(BytesIO(zipresp.read())) as zfile:
+                    zfile.extractall("data/demand/unsd/data")
+
+                    path = "data/demand/unsd/data"
+
+    # Get the files from the path provided in the OP
+    all_files = Path("data/demand/unsd/data").glob("*.txt")
+
+    # Create a dataframe from all downloaded files
+    df = pd.concat(
+        (pd.read_csv(f, encoding="utf8", sep=";") for f in all_files), ignore_index=True
+    )
+
+    # Split 'Commodity', 'Transaction' column to two
+    df[["Commodity", "Transaction", "extra"]] = df["Commodity - Transaction"].str.split(
+        " - ", expand=True
+    )
+
+    # Remove Foootnote and Estimate from 'Commodity - Transaction' column
+    df = df.loc[df["Commodity - Transaction"] != "Footnote"]
+    df = df.loc[df["Commodity - Transaction"] != "Estimate"]
+
+    # Create a column with iso2 country code
+    cc = coco.CountryConverter()
+    Country = pd.Series(df["Country or Area"])
+
+    df["country"] = cc.pandas_convert(series=Country, to="ISO2", not_found="not found")
+
+    # remove countries or areas that have no iso2 such as former countries names
+    df = df.loc[df["country"] != "not found"]
+
+    # Convert country column that contains lists for some country names that are identified with more than one country.
+    df["country"] = df["country"].astype(str)
+
+    # Remove all iso2 conversions for some country names that are identified with more than one country.
+    df = df[~df.country.str.contains(",", na=False)].reset_index(drop=True)
+
+    # Create a dictionary with all the conversion factors from ktons or m3 to TWh based on https://unstats.un.org/unsd/energy/balance/2014/05.pdf
+    fuels_conv_toTWh = {
```

Comment:
Maybe move to config. Although this is nothing that should be changed...

Reply:
I thought about it, but I think the config is long enough. This will make it longer unnecessarily. We can add separate files for this and the colors and so on once we have some time.